q2_types_genomics/feature_map/tests/test_transformer.py

```diff
@@ -0,0 +1,47 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import json
+import unittest
+
+from qiime2.plugin.testing import TestPluginBase
+
+from q2_types_genomics.feature_map import MAGtoContigsFormat
+
+
+class TestTransformers(TestPluginBase):
+    package = "q2_types_genomics.feature_map.tests"
+
+    def setUp(self):
+        super().setUp()
+        self.valid_contig_map = {
```

Comment:
Hey @gregcaporaso, two thoughts here:
- I did not include a contig with a space in the name as it's impossible to have those in FASTA files; the only way for this to happen is when someone would edit the IDs by hand (which I find highly unlikely, considering the size of those datasets) - so my conclusion was not to include this check as it adds unnecessary processing of _a lot_ of IDs (for large datasets) - if you think, though, that it should be there  I'll add it in the validator
- similarly, I'm not testing whether there are duplicated IDs ad getting those is pretty much impossible either; the only scenario when it would happen is when user generated bins externaly using a bin refinement tool - the most common tools do make sure though that no contig ID is duplicated in each bin

This kind of comes down to the question, whether we want to safeguard against users importing data which was manually edited (and this, in the particular case of shotgun analysis is very unlikely, I guess)... What do you think? I'm happy to add the checks, but I don't want to add additional runtime whenever is not really necessary.

Reply:
yeah these sound like edge cases that are unlikely to ever occur.

Reply:
Sounds good - both of those suggestions were "if they were allowed", and it sounds like they are effectively not. I do still recommend varying the lengths of the lists of contigs.